Libraries/LibWeb/Fetch/Infrastructure/FetchController.cpp

```diff
@@ -96,7 +96,27 @@ void FetchController::abort(JS::Realm& realm, Optional<JS::Value> error)
     m_serialized_abort_reason = structured_serialize(realm.vm(), error.value(), fallback_error);
 }
 
-// FIXME: https://fetch.spec.whatwg.org/#deserialize-a-serialized-abort-reason
+// https://fetch.spec.whatwg.org/#deserialize-a-serialized-abort-reason
+GC::Ref<WebIDL::DOMException> FetchController::deserialize_a_serialized_abort_reason(JS::Realm& realm)
+{
+    // 1. Let fallbackError be an "AbortError" DOMException.
+    auto fallback_error = WebIDL::AbortError::create(realm, "Fetch was aborted"_string);
+
+    // 2. Let deserializedError be fallbackError.
+    auto deserialized_error = fallback_error;
+
+    // 3. If abortReason is non-null, then set deserializedError to StructuredDeserialize(abortReason, realm).
+    //    If that threw an exception or returned undefined, then set deserializedError to fallbackError.
+    if (m_serialized_abort_reason.has_value()) {
+        auto deserialized_error_or_exception = HTML::structured_deserialize(realm.vm(), m_serialized_abort_reason.value(), realm, {});
+        if (!deserialized_error_or_exception.is_exception() && !deserialized_error_or_exception.value().is_undefined()) {
+            deserialized_error = verify_cast<WebIDL::DOMException>(deserialized_error_or_exception.value().as_object());
+        }
```

Comment:
Do we know for sure that a FetchController can only be aborted with a DOMException?

Reply:
You are right, it can actually be any JS::Value.